frontend/src/modules/_shared/DataProviderFramework/visualization/hooks/useSubscribedProviderHoverVisualizations.ts

```diff
@@ -17,93 +18,114 @@ export type AssemblerProviderHoverVisualizations<TTarget extends VisualizationTa
+                    });
+                });
+
+                unsubscribeFunctions.push(topicUnsubFunc);
             }
 
             return function unsubscribeFromWorkbenchServices() {
                 for (const unsubscribeFunction of unsubscribeFunctions) {
                     unsubscribeFunction();
                 }
-                setVisualizations([]);
+                setHoverData({});
             };
         },
-        [visualizationAssemblerProduct, workbenchServices],
+        [hoverService, moduleInstanceId, visualizationAssemblerProduct, visualizationFunctions],
     );
 
-    return visualizations.map((visualization) => ({
-        groupId: visualization.groupId,
-        hoverVisualizations: Object.values(visualization.hoverVisualizations).flat(),
-    }));
+    return React.useMemo(() => {
+        const visualizations = [] as AssemblerProviderHoverVisualizations<TTarget>[];
+
+        for (const visualizationFunc of visualizationFunctions) {
+            const hoverFunctions = visualizationFunc.hoverVisualizationFunctions;
+            const groupVisualization = [];
+
+            for (const untypedTopic in hoverFunctions) {
+                const { topic, topicVisualization } = getTypedTopicAndFunc(untypedTopic, hoverFunctions);
+
+                if (topic && topicVisualization) {
+                    const visualization = topicVisualization(hoverData[topic] ?? null);
+                    groupVisualization.push(visualization);
+                }
+            }
+
+            visualizations.push({
+                groupId: visualizationFunc.groupId,
+                hoverVisualizations: groupVisualization.flat(),
+            });
+        }
+        return visualizations;
+    }, [hoverData, visualizationFunctions]);
+}
+
+function getTypedTopicAndFunc<TTarget extends VisualizationTarget>(
+    untypedTopic: string,
+    functions: HoverVisualizationFunctions<TTarget>,
+) {
+    if (!Object.values<string>(HoverTopic).includes(untypedTopic)) return {};
+
+    const typedTopic = untypedTopic as HoverTopic;
+    type TFunc = HoverVisualizationFunction<TTarget, typeof typedTopic>;
+    const typedFunc = functions[typedTopic] as TFunc | undefined;
+
+    return {
+        topic: typedTopic,
+        topicVisualization: typedFunc,
+    };
 }
 
 function flattenVisualizationFunctionsRecursively<TTarget extends VisualizationTarget>(
     visualizationGroup: VisualizationGroup<TTarget> | AssemblerProduct<TTarget, any, any>,
 ): InternalAssemblerProviderHoverVisualizationFunctions<TTarget>[] {
     const visualizationFunctions: InternalAssemblerProviderHoverVisualizationFunctions<TTarget>[] = [];
-    if (visualizationGroup.hoverVisualizationFunctions) {
+    const groupId = Object.hasOwn(visualizationGroup, "id")
+        ? (visualizationGroup as VisualizationGroup<TTarget>).id
+        : undefined;
+
+    if (visualizationGroup.hoverVisualizationFunctions && groupId !== "") {
```

Comment:
Is it allowed with `groupId === undefined` if so, add comment for clarification. Otherwise add comment to state explicit checking for "valid" id string (i.e. non-empty)